### PR TITLE
setting-up-investment-calculator-chart

### PR DESCRIPTION
**Pull request by jvin08**

### CODE REVIEW

File: app/components/investCalculator/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import { selectDarkmode, selectCurrency } from "@/app/lib/dynamicValuesSlice";
+import clsx from "clsx";
+
+export const Header = ({price, priceDate } : {price: number, priceDate: number}) => {
+  const darkmode = useSelector(selectDarkmode);
+  const currency = useSelector(selectCurrency);
+  const coinLastPrice = currency.sign + Number(price).toFixed(3);
+  const dateNow = new Date(priceDate).toLocaleString("default", { month: "long" }).slice(0,3) + " " + new Date(priceDate).getDate() + ", " + new Date(priceDate).getUTCFullYear();
```
Comment (yoloOnTheBattlefield) [inline app/components/investCalculator/Header.tsx]: const day = new Date(priceDate)
const dateNow = day.toLocaleString("default", { month: "long" }).slice(0,3) + " " + day.getDate() + ", " + day.getUTCFullYear();

File: app/components/investCalculator/SpentAmount.tsx [hunk 2]
```diff
@@ -19,10 +20,13 @@ const SpentAmount = (
     days: number | undefined
     startTime: string,
     endTime: string
+    getChartData: Function
   }) => {
   const { data } = useGetCoinDataByDateQuery(query);
   const coinPrices = filterPrices(data, days, interval, startTime, endTime);
   const spentMoney = amountInvested(initialAmount, coinPrices, growRate);
+  const chartData = spentMoney[2] as number[][];
+  getChartData(chartData);
```
Comment (yoloOnTheBattlefield) [inline app/components/investCalculator/SpentAmount.tsx]: useEffect(() => {getChartData(chartData);} )

Comment (yoloOnTheBattlefield) [reply app/components/investCalculator/SpentAmount.tsx]: [chartData]

Comment (yoloOnTheBattlefield) [reply app/components/investCalculator/SpentAmount.tsx]: put the chartData as dependency

File: app/components/investCalculator/SpentAmountDCA.tsx [hunk 3]
```diff
@@ -10,19 +10,23 @@ const SpentAmountDCA = (
...
     startTime: string,
-    endTime: string
+    endTime: string,
+    getChartData: Function
   }) => {
   const { data } = useGetCoinDataByDateQuery(query);
   const coinPrices = filterPrices(data, days, interval, startTime, endTime);
   const spentMoney = amountInvestedDCA(initialAmount, coinPrices, growAmount);
+  const chartData = spentMoney[2] as number[][];
+  getChartData(chartData);
```
Comment (yoloOnTheBattlefield) [inline app/components/investCalculator/SpentAmountDCA.tsx]: same here

File: app/components/investCalculator/ValueCostAverage.tsx [hunk 4]
```diff
@@ -45,11 +48,35 @@ const ValueCostAverage = ({coin}:{coin: string}) => {
...
+                fill={"#1fcac0"}  fillOpacity={1} />
+            </svg>
+            </div> 
+            : <span className="absolute top-1 left-8 text-lg">☶</span>
+          }</button>
```
Comment (yoloOnTheBattlefield) [inline app/components/investCalculator/ValueCostAverage.tsx:79]: make it a reusable component

File: app/components/investCalculator/options.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      grid: {
+        display: false, // Hide grid lines on X-axis
+      },
+    },
+  }, 
+};
+export const getChartData = (data: number[][]) => {
+  const borderWidth = 1.5;
+  const time = data.map((item:number[])=>formatStandardDate(new Date(item[1])));
+  const prices = data.map((item:number[])=>item[0]);
```
Comment (yoloOnTheBattlefield) [inline app/components/investCalculator/options.ts]: const [time, prices] = data.reduce(
  (acc, [price, time]) => ({
    time: [...acc.time, formatStandardDate(new Date(time))],
    prices: [...acc.prices, price]
  }),
  { time: [], prices: [] }
);

File: app/components/investCalculator/utils.ts [hunk 6]
```diff
@@ -5,30 +5,32 @@ export const timeInterval = (purchaseDate: string) => {
...
-    const actualRate = coinPrices[i] / coinPrices[i-1];
+    const actualRate = coinPrices[i][1] / coinPrices[i-1][1];
     const newValue = growArray[i-1] * actualRate;
     spentAmount += plannedNewValue - newValue;
     growArray.push(plannedNewValue);
     actualGrowArray.push(newValue);
   }
-  return [Math.floor(spentAmount), Math.floor(actualGrowArray.slice(-1)?.[0])];
+  const growArrayWithTime = actualGrowArray.map((item, index) => [item, coinPrices?.[index][0]]);
+  return [Math.floor(spentAmount), Math.floor(actualGrowArray.slice(-1)?.[0]), growArrayWithTime];
```
Comment (yoloOnTheBattlefield) [inline app/components/investCalculator/utils.ts]: return {
spentAmount: Math.floor(spentAmount),
lastValue: Math.floor(actualGrowArray.slice(-1)?.[0]),
totalInvestements: growArrayWithTime
}

File: app/components/investCalculator/utils.ts [hunk 7]
```diff
@@ -5,30 +5,32 @@ export const timeInterval = (purchaseDate: string) => {
...
-    const actualGrow = coinPrices[i] / coinPrices[i-1];
+    const actualGrow = coinPrices[i][1] / coinPrices[i-1][1];
     const newValue = growArray[i-1] * actualGrow + Number(growAmount);
     spentAmount += Number(growAmount);
     growArray.push(newValue);
   }
-  return [Math.floor(spentAmount), Math.floor(growArray.slice(-1)[0])];
+  const growArrayWithTime = growArray.map((item, index) => [item, coinPrices?.[index][0]]);
+  return [Math.floor(spentAmount), Math.floor(growArray.slice(-1)[0]), growArrayWithTime];
 };
```
Comment (yoloOnTheBattlefield) [inline app/components/investCalculator/utils.ts:42]: same here

**Review by yoloOnTheBattlefield — DISMISSED** [review]

**Review by jvin08 — COMMENTED** [review]
fixed

**Pull request merged**